File: frontend/src/types/market.ts
export type MarketSymbol = 'SPX' | 'NDX' | 'TNX'

export interface MarketIndexItem {
  symbol: MarketSymbol
  name: string
  close: number
  prevClose: number | null
  changePct: number | null
  date: string
}

export interface BreakoutItem {
  ticker: string
  companyName: string
  closePrice: number
  ma150Value: number
  pctAboveMa150: number
  marketCap: number
}

export interface BreakoutSnapshot {
  scanDate: string | null
  scannedAt: string | null
  items: BreakoutItem[]
  total: number
}

File: backend/app/schemas/market.py
from __future__ import annotations

from datetime import date, datetime

from app.schemas.watchlist import CamelModel


class MarketIndexOut(CamelModel):
    symbol: str
    name: str
    close: float
    prev_close: float | None = None
    change_pct: float | None = None
    date: date


class BreakoutItemOut(CamelModel):
    ticker: str
    company_name: str
    close_price: float
    ma150_value: float
    pct_above_ma150: float
    market_cap: int


class BreakoutSnapshotOut(CamelModel):
    scan_date: date | None
    scanned_at: datetime | None
    items: list[BreakoutItemOut]
    total: int
